lib/DNS_Cache.cpp

```
@@ -0,0 +1,328 @@
+	std::string result;
+	char ip_addr[INET6_ADDRSTRLEN];
+
+	union {
+		struct sockaddr_in in;
+		struct sockaddr_in6 in6;
+	} custom_sockaddr;
+
+	struct sockaddr* addr = (struct sockaddr*)malloc(sizeof(custom_sockaddr));
+	socklen_t addrlen = sizeof(custom_sockaddr);
+	memset(addr, 0, sizeof(custom_sockaddr));
+
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)